unified_planning/engines/compilers/trajectory_constraints_remover.py

```diff
@@ -0,0 +1,422 @@
+
+    @staticmethod
+    def supports_compilation(compilation_kind: CompilationKind) -> bool:
+        return compilation_kind == CompilationKind.TRAJECTORY_CONSTRAINTS_REMOVING
+
+    @staticmethod
+    def supported_kind() -> ProblemKind:
+        supported_kind = ProblemKind()
+        supported_kind.set_problem_class("ACTION_BASED")
+        supported_kind.set_typing("FLAT_TYPING")
+        supported_kind.set_typing("HIERARCHICAL_TYPING")
+        supported_kind.set_numbers("CONTINUOUS_NUMBERS")
+        supported_kind.set_numbers("DISCRETE_NUMBERS")
+        supported_kind.set_fluents_type("NUMERIC_FLUENTS")
+        supported_kind.set_fluents_type("OBJECT_FLUENTS")
+        supported_kind.set_conditions_kind("NEGATIVE_CONDITIONS")
+        supported_kind.set_conditions_kind("DISJUNCTIVE_CONDITIONS")
+        supported_kind.set_conditions_kind("EQUALITY")
+        supported_kind.set_conditions_kind("EXISTENTIAL_CONDITIONS")
+        supported_kind.set_conditions_kind("UNIVERSAL_CONDITIONS")
+        supported_kind.set_effects_kind("CONDITIONAL_EFFECTS")
+        supported_kind.set_effects_kind("INCREASE_EFFECTS")
+        supported_kind.set_effects_kind("DECREASE_EFFECTS")
+        supported_kind.set_time("CONTINUOUS_TIME")
+        supported_kind.set_time("DISCRETE_TIME")
+        supported_kind.set_time("INTERMEDIATE_CONDITIONS_AND_EFFECTS")
+        supported_kind.set_time("TIMED_EFFECT")
+        supported_kind.set_time("TIMED_GOALS")
+        supported_kind.set_time("DURATION_INEQUALITIES")
+        supported_kind.set_simulated_entities("SIMULATED_EFFECTS")
+        supported_kind.set_constraints_kind("TRAJECTORY_CONSTRAINTS")
+        return supported_kind
+
+    def _compile(
+        self,
+        problem: "up.model.AbstractProblem",
+        compilation_kind: "up.engines.CompilationKind",
+    ) -> CompilerResult:
+        """
+        Takes an instance of a :class:`~unified_planning.model.Problem` and the `TRAJECTORY_CONSTRAINTS_REMOVING` :class:`~unified_planning.engines.CompilationKind`
+        and returns a `CompilerResult` where the problem whitout trajectory_constraints.
+
+        :param problem: The instance of the `Problem` that contains the trajecotry constraints.
+        :param compilation_kind: The `CompilationKind` that must be applied on the given problem;
+            only `TRAJECTORY_CONSTRAINTS_REMOVING` is supported by this compiler
+        :return: The resulting `CompilerResult` data structure.
+        """
+        assert isinstance(problem, Problem)
+        env = problem.env
+        substituter = Substituter(env)
+        expression_quantifier_remover = ExpressionQuantifiersRemover(problem.env)
+        self._grounding_result = engines.compilers.grounder.Grounder().compile(
+            problem, CompilationKind.GROUNDING
+        )
+        assert isinstance(self._grounding_result.problem, Problem)
+        self._problem = self._grounding_result.problem.clone()
```

Comment:
Why does this need to be a field?

(In general, delcaring fields in methods other than `__init__` is a bad idea...)